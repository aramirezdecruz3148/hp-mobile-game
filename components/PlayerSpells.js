import React from "react";
import { SafeAreaView, FlatList, Text, View} from "react-native";
import PropTypes from "prop-types";

function Spell({ spell, type, effect }) {
  return (
    <View>
      <Text>{spell}</Text>
      <Text>{type}</Text>
      <Text>{effect}</Text>
    </View>
  );
}

Spell.propTypes = {
  spell: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  effect: PropTypes.string.isRequired
}

export default function PlayerSpells({ spellArray }) {
  return (
    <SafeAreaView>
      <FlatList 
        data={spellArray}
        renderItem={({ item }) => (<Spell 
          spell={item.spell} 
          type={item.type} 
          effect={item.effect}
          keyExtractor={item.spell} 
        />)}
      />
    </SafeAreaView>
  );
}

PlayerSpells.propTypes = {
  spellArray: PropTypes.arrayOf(PropTypes.shape({
    spell: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    effect: PropTypes.string.isRequired
  })).isRequired
}