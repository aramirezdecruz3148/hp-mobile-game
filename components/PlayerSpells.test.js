import React from 'react';
import { shallow } from 'enzyme';
import PlayerSpells from './PlayerSpells';

describe('PlayerSpells component', () => {
  it('renders PlayerSpells', () => {
    const wrapper = shallow(<PlayerSpells 
      spellArray={[
        {
          spell: "Anteoculatia",
          type: "Hex",
          effect: "turns opponent's hair into antlers",
          damage: -1,
          outcome: `You were hit with the Anteoculatia hex, which caused
          giant antlers to sprout on your head and took 1 damage point from
          your health`
        },
        {
          spell: "Avis",
          type: "Spell",
          effect: "launches birds at your opponent",
          damage: -2,
          outcome: `You were hit with the Avis spell, which caused birds to swarm
          and attack you and took 2 damage points from your health`
        },
        {
          spell: "Babbling Curse ",
          type: "Curse",
          effect: "makes opponent endlessly babble",
          damage: -1,
          outcome: `You were hit with the Babbling curse, which is caused you to 
          endlessly babble none sense and took 2 damage points from your health`
        }
      ]} 
      />);
    expect(wrapper).toMatchSnapshot();
  });
});