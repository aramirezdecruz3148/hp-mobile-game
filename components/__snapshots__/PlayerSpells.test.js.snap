// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSpells component renders PlayerSpells 1`] = `
<SafeAreaView>
  <FlatList
    data={
      Array [
        Object {
          "damage": -1,
          "effect": "turns opponent's hair into antlers",
          "outcome": "You were hit with the Anteoculatia hex, which caused
          giant antlers to sprout on your head and took 1 damage point from
          your health",
          "spell": "Anteoculatia",
          "type": "Hex",
        },
        Object {
          "damage": -2,
          "effect": "launches birds at your opponent",
          "outcome": "You were hit with the Avis spell, which caused birds to swarm
          and attack you and took 2 damage points from your health",
          "spell": "Avis",
          "type": "Spell",
        },
        Object {
          "damage": -1,
          "effect": "makes opponent endlessly babble",
          "outcome": "You were hit with the Babbling curse, which is caused you to 
          endlessly babble none sense and took 2 damage points from your health",
          "spell": "Babbling Curse ",
          "type": "Curse",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</SafeAreaView>
`;
